lib/puppet/rest/client.rb

```diff
@@ -67,25 +65,11 @@ def put(url, body:, query: nil, header: nil)
 
     private
 
-    # Checks for SSL certificates on disk and sets VERIFY_PEER
-    # if they are found. Otherwise, sets VERIFY_NONE.
-    def configure_verify_mode(ssl_config)
-      # Either the path to an external CA or to our CA cert from the Puppet master
-      # TODO We may be able to consolidate this with the current intermediate CA work?
+    def configure_verify_mode(ssl_context)
       ca_path = Puppet[:ssl_client_ca_auth] || Puppet[:localcacert]
-
-      if ssl_certificates_are_present?(ca_path)
-        ssl_config.verify_mode = OpenSSL::SSL::VERIFY_PEER
-        ssl_config.add_trust_ca(ca_path)
-        ssl_config.verify_callback = Puppet::SSL::Validator::DefaultValidator.new(ca_path)
-        ssl_config.set_client_cert_file(Puppet[:hostcert], Puppet[:hostprivkey])
-      else
-        ssl_config.verify_mode = OpenSSL::SSL::VERIFY_NONE
-      end
-    end
-
-    def ssl_certificates_are_present?(ca_path)
-      Puppet::FileSystem.exist?(Puppet[:hostcert]) && Puppet::FileSystem.exist?(ca_path)
```

Comment:
Why did we need to check these certificates were present previously, but not now?

Reply:
Because now we are relying on the caller to add them to the SSLContext that is passed to this client when it is created. The caller (in this case SSL::Host) has already checked for their existence before it tries to make a request.